discord_bots/cogs/queue.py

```diff
@@ -0,0 +1,76 @@
+from discord.ext.commands import Bot, Context, check, command
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.orm.exc import NoResultFound
+
+from discord_bots.checks import is_admin
+from discord_bots.cogs.base import BaseCog
+from discord_bots.models import Map, Queue, Rotation, RotationMap
+
+
+class QueueCog(BaseCog):
+    def __init__(self, bot: Bot):
+        super().__init__(bot)
+
+    @command()
+    @check(is_admin)
+    async def setqueuerotation(self, ctx: Context, queue_name: str, rotation_name: str):
+        session = ctx.session
+
+        try:
+            queue = session.query(Queue).filter(Queue.name.ilike(queue_name)).one()
+        except NoResultFound:
+            await self.send_error_message(f"Could not find queue **{queue_name}**")
+            return
+
+        try:
+            rotation = (
+                session.query(Rotation).filter(Rotation.name.ilike(rotation_name)).one()
+            )
+        except NoResultFound:
+            await self.send_error_message(
+                f"Could not find rotation **{rotation_name}**"
+            )
+            return
+
+        queue.rotation_id = rotation.id
+        session.commit()
+        await self.send_success_message(
+            f"Rotation for **{queue.name}** set to **{rotation.name}**"
+        )
+
+    @command()
+    async def showqueuerotation(self, ctx: Context, queue_name: str):
+        session = ctx.session
+
+        queue_data = (
+            session.query(
+                Queue.name, Rotation.name, RotationMap.ordinal, Map.short_name
+            )
+            .filter(Queue.name.ilike(queue_name))
+            .outerjoin(Rotation, Rotation.id == Queue.rotation_id)
+            .outerjoin(RotationMap, RotationMap.rotation_id == Rotation.id)
+            .outerjoin(Map, RotationMap.map_id == Map.id)
+            .order_by(RotationMap.ordinal.asc())
+            .all()
+        )
+
+        if not queue_data:
+            await self.send_error_message(f"Could not find queue **{queue_name}**")
+            return
+
+        if not queue_data[0][1]:
+            await self.send_error_message(
+                f"Queue **{queue_data[0][0]}** has not been assigned a rotation"
+            )
+            return
+
+        maps = []
+        if not queue_data[0][3]:
+            maps = ["None"]
+        else:
+            for row in queue_data:
+                maps.append(row[3])
+
+        output = f"**{queue_data[0][0]}** is assigned to **{queue_data[0][1]}**\n"
```

Comment:
reposting this from above, but here's what queue_data looks like:
`[('LTgold', 'Global', 1, 'DX'), ('LTgold', 'Global', 2, 'HB')]`
so maybe I can set queue_name and rotation_name.  not really sure how to make the queue_data[0][3] part more readable.  maybe I should just do separate queries for queue, rotation, and the maps?  several times I was torn about doing multiple queries like that - I always find it more readable, but I also know it's more efficient to get everything you need in a single query.  I'm not really sure how big that tradeoff is though.  maybe almost nonexistent since each of these tables is so small?

also, I guess I don't need to select ordinal here.  I thought you needed to select it to order by it, but it seems like you don't.

Reply:
ahh i understand now, thx for explaining

ok this is def nbd and feel feel not to change, but if you're torn on how to decide, its generally accepted to bias for readability over performance initially